File: Java/CreationalDesignPattern/src/main/java/com/ute/refactoring/abstractfactory/factories/WindowsFactory.java
package com.ute.refactoring.abstractfactory.factories;

import com.ute.refactoring.abstractfactory.buttons.Button;
import com.ute.refactoring.abstractfactory.buttons.WindowsButton;
import com.ute.refactoring.abstractfactory.checkboxes.Checkbox;
import com.ute.refactoring.abstractfactory.checkboxes.WindowsCheckbox;

public class WindowsFactory implements GUIFactory{
    @Override
    public Button createButton() {
        return new WindowsButton();
    }

    @Override
    public Checkbox createCheckbox() {
        return new WindowsCheckbox();
    }
}

File: C#/CreationalDesignPattern/CreationalDesignPattern/Refactoring/AbstractFactory/Factories/WindowsFactory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CreationalDesignPattern.Refactoring.AbstractFactory
{
    public class WindowsFactory : GUIFactory
    {
        public IButton CreateButton()
        {
            return new WindowsButton();
        }

        public Checkbox CreateCheckbox()
        {
            return new WindowsCheckbox();
        }
    }
}
